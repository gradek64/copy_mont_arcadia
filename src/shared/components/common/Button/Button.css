/** @define Button */

@define-mixin buttonColor $color: null, $bgcolor: null, $bordercolor: null {
  background-color: $bgcolor;
  border: $cta-border-width solid $bordercolor;
  color: $color;
  @mixin-content;
}

.Button {
  display: block;
  font-family: $font-family-button;
  width: 100%;
  min-height: $form-element-height;
  padding: 5px;
  margin: 0;
  border: $cta-border-width solid $cta-primary-border-color;
  border-radius: $cta-border-radius;
  background-color: $cta-primary-background;
  color: $cta-primary-font-color;
  font-size: $font-size-button;
  font-weight: $font-weight-button;
  text-align: center;
  text-transform: $cta-primary-text-transform;
  touch-action: manipulation;
  cursor: pointer;
  box-shadow: $cta-box-shadow;
  letter-spacing: $cta-letter-spacing;

  &.is-hidden {
    display: none;
  }

  &.is-active {
    @mixin buttonColor $cta-primary-is-active-font-color, $cta-primary-is-active-background, $cta-primary-is-active-border-color {
      outline: 0;
    }
  }

  &[disabled] {
    @mixin buttonColor $cta-primary-inactive-font-color, $cta-primary-inactive-background, $cta-primary-inactive-border-color {}

    &:hover {
      cursor: default;
    }
  }

  &--secondary {
    @mixin buttonColor $cta-secondary-font-color, $cta-secondary-background, $cta-secondary-border-color {
      text-transform: $cta-secondary-text-transform;
      font-size: $cta-secondary-font-size;

    }

    &[disabled] {
      @mixin buttonColor $cta-secondary-inactive-font-color, $cta-secondary-inactive-background, $cta-secondary-inactive-border-color {}
    }
  }

  &--tertiary {
    @mixin buttonColor $cta-tertiary-font-color, $cta-tertiary-background, $cta-tertiary-border-color {
      text-transform: $cta-tertiary-text-transform;
      text-decoration: $cta-tertiary-text-decoration;
      text-align: $cta-tertiary-text-align;
      width: $cta-tertiary-width;
      font-size: $cta-tertiary-font-size;
    }

    &[disabled] {
      @mixin buttonColor $cta-tertiary-inactive-font-color, $cta-tertiary-inactive-background, $cta-tertiary-inactive-border-color {}
    }
  }

  &--link {
    text-decoration: underline;
  }

  &--fullWidth {
    width: 100%;
  }

  &--halfWidth {
    flex: 1 0;
    margin-right: $cta-gutter;
  }

  &--threeFifthWidth {
    flex: 3 0;
    margin-right: $cta-gutter;
  }

  &--twoFifthWidth {
    flex: 2 0;
    margin-right: $cta-gutter;
  }

  &--linkButton {
    text-decoration: none;
    line-height: 1.9em;
    min-height: inherit;
    width: inherit;
  }

  &--isDisabled {
    @mixin buttonColor $cta-primary-inactive-font-color, $cta-primary-inactive-background, $cta-primary-inactive-border-color {}

    cursor: not-allowed
  }

}

@media $min-laptop {
  .Button {
    &:enabled:hover {
      @mixin buttonColor $cta-primary-hover-font-color, $cta-primary-hover-background, $cta-primary-hover-border-color {
        font-weight: $cta-primary-hover-font-weight;
      }
    }
    &.is-active:enabled:hover {
      @mixin buttonColor $cta-primary-is-active-hover-font-color, $cta-primary-is-active-hover-background, $cta-primary-is-active-hover-border-color {
        outline: 0;
      }
    }
    &-secondary {
      &:enabled:hover {
        @mixin buttonColor $cta-secondary-hover-font-color, $cta-secondary-hover-background, $cta-secondary-hover-border-color {
          font-weight: $cta-secondary-hover-font-weight;
        }
      }
    }
    &-tertiary {
      &:enabled:hover {
        @mixin buttonColor $cta-tertiary-hover-font-color, $cta-tertiary-hover-background, $cta-tertiary-hover-border-color {
          font-weight: $cta-tertiary-hover-font-weight;
        }
      }
    }
  }
}

/* postcss-bem-linter: ignore */
.no-touch .Button:enabled:hover {
 @mixin buttonColor $cta-primary-hover-font-color, $cta-primary-hover-background, $cta-primary-hover-border-color {
   font-weight: $cta-primary-hover-font-weight;
 }
}
/* postcss-bem-linter: ignore */
.no-touch .Button:enabled:active {
  @mixin buttonColor $cta-primary-active-font-color, $cta-primary-active-background, $cta-primary-active-border-color {
    font-weight: $cta-primary-active-font-weight;
    outline: 0;
  }
}
/* postcss-bem-linter: ignore */
.no-touch .Button.is-active:enabled:hover {
  @mixin buttonColor $cta-primary-is-active-hover-font-color, $cta-primary-is-active-hover-background, $cta-primary-is-active-hover-border-color {
    font-weight: $cta-primary-active-font-weight;
    outline: 0;
  }
}
/* postcss-bem-linter: ignore */
.no-touch .Button-secondary:enabled:hover {
 @mixin buttonColor $cta-secondary-hover-font-color, $cta-secondary-hover-background, $cta-secondary-hover-border-color {
   font-weight: $cta-secondary-hover-font-weight;
 }
}
/* postcss-bem-linter: ignore */
.no-touch .Button-secondary:enabled:active {
  @mixin buttonColor $cta-secondary-active-font-color, $cta-secondary-active-background, $cta-secondary-active-border-color {
    font-weight: $ cta-secondary-active-font-weight;
  }
}
/* postcss-bem-linter: ignore */
.no-touch .Button-tertiary:enabled:hover {
 @mixin buttonColor $cta-tertiary-hover-font-color, $cta-tertiary-hover-background, $cta-tertiary-hover-border-color {
   font-weight: $cta-tertiary-hover-font-weight;
 }
}
/* postcss-bem-linter: ignore */
.no-touch .Button-tertiary:enabled:active {
  @mixin buttonColor $cta-tertiary-active-font-color, $cta-tertiary-active-background, $cta-tertiary-active-border-color {
    font-weight: $cta-tertiary-active-font-weight;
  }
}
