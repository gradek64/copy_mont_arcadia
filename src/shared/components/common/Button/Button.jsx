import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

export default class Button extends Component {
  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
    isDisabled: PropTypes.bool,
    isActive: PropTypes.bool,
    clickHandler: PropTypes.func,
    type: PropTypes.string,
    ariaLabel: PropTypes.string,
    lang: PropTypes.string,
  }

  handleClick = (e) => {
    const { isDisabled, clickHandler } = this.props
    if (!isDisabled && clickHandler) clickHandler(e)
  }

  render() {
    const {
      isDisabled,
      isActive,
      type,
      className,
      ariaLabel,
      children,
      lang,
    } = this.props
    const classNames = classnames('Button', className, {
      'is-active': isActive,
      // ADP-3192 - add to bag button breaks when text is translated
      // If React is not text only element, add notranslate to prevent google
      // translate from breaking children of button component
      // any children passed to button with text nodes should add className='...classes translate'
      // to the text nodes to reenable translation on those elements.
      // See this for more info: https://github.com/facebook/react/issues/11538
      notranslate: typeof children !== 'string',
    })

    return (
      <button
        className={classNames}
        type={type || 'button'}
        role="button"
        disabled={isDisabled}
        aria-disabled={isDisabled}
        aria-hidden={isDisabled}
        aria-label={ariaLabel}
        onClick={this.handleClick}
        lang={lang}
      >
        {children}
      </button>
    )
  }
}
