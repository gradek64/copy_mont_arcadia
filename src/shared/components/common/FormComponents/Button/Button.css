/** @define ButtonContainer */

/* postcss-bem-linter: ignore */
.ButtonContainer .Input-validationMessage {
  display: block;
  margin-bottom: 5px;
}
