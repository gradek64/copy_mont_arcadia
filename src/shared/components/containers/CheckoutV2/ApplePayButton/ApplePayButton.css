/** @define ApplePayButton; weak */
.ApplePayButton {
  width: 100%;
  height: 42px;
  background-color: black;
  color: white;

  &--button {
    background-color: black;
    height: 100%;
    width: 100%;

    &.is-active{
      background-color: black;
      border: 0px;
    }
  }


}

@supports not (-webkit-appearance: -apple-pay-button) {
  .ApplePayButton--button {
    display: inline-flex;
    justify-content: center;
    font-size: 12px;
    border-radius: 5px;
    padding: 0px;
    box-sizing: border-box;
    min-width: 200px;
    min-height: 32px;
    max-height: 64px;

    background: #000;
    border-radius: 0px;

    &>.logo {
      background-image: -webkit-named-image(apple-pay-logo-white);
      background-color: black;
      width: 35px;
      height: 100%;
      background-size: 100% 60%;
      background-repeat: no-repeat;
      background-position: 0 50%;
      margin-left: 2px;
      border: none;
    }

    &>.text {
      font-family: -apple-system;
      font-size: 1em;
      font-weight: 300;
      align-self: center;
      margin-right: 2px;
    }
  }
}

@supports (-webkit-appearance: -apple-pay-button) {
  .ApplePayButton--button  {
    display: inline-block;
    -webkit-appearance: -apple-pay-button;
    -apple-pay-button-type: buy;
    -apple-pay-button-style: black;

    &>* {
      display: none;
    }
  }
}