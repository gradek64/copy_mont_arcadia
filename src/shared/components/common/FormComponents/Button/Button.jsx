import React from 'react'
import classnames from 'classnames'
import PropTypes from 'prop-types'

// Components
import Button from '../../Button/Button'
import InputError from '../InputError/InputError'

const FormButton = (
  {
    name,
    error = '',
    isDisabled = false,
    type = 'button',
    onClick,
    children,
    customClass,
    touched = false,
  },
  { l }
) => {
  const componentClassName = classnames(
    'ButtonContainer',
    {
      [`ButtonContainer-${name}`]: !!name,
    },
    {
      [`FormComponent-${name}`]: !!name,
    },
    customClass
  )

  return (
    <div className={componentClassName}>
      {!isDisabled && touched && error ? (
        <InputError name={name} className="Input-validationMessage">
          {l(error)}
        </InputError>
      ) : null}
      <Button isDisabled={isDisabled} type={type} clickHandler={onClick}>
        {children}
      </Button>
    </div>
  )
}

FormButton.propTypes = {
  name: PropTypes.string.isRequired,
  error: PropTypes.string,
  isDisabled: PropTypes.bool,
  type: PropTypes.string,
  onClick: PropTypes.func,
  customClass: PropTypes.string,
}

FormButton.contextTypes = {
  l: PropTypes.func,
}

export default FormButton
