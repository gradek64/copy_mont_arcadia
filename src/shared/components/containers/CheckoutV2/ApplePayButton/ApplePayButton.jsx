import PropTypes from 'prop-types'
import React, { Component } from 'react'
import Button from '../../../common/Button/Button'

class ApplePayButton extends Component {
  render() {
    const { clickHandler, isActive, isDisabled, lang } = this.props
    const { l } = this.context

    return (
      <div className="ApplePayButton">
        <Button
          lang={lang}
          className="ApplePayButton--button"
          clickHandler={clickHandler}
          isActive={isActive}
          isDisabled={isDisabled}
        >
          {/*
      According to the ApplePay doc, old versions of Safari don't support -webkit-appearance: -apple-pay-button,
      which will handle to show the button and translate the text.
      The html below is to support old versions of Safari.
      */}
          <span className="text">{l`Buy with`}</span>
          <span className="logo" />
        </Button>
      </div>
    )
  }
}

ApplePayButton.propTypes = {
  clickHandler: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired,
  isDisabled: PropTypes.bool.isRequired,
  lang: PropTypes.string.isRequired,
}

ApplePayButton.contextTypes = {
  l: PropTypes.func,
}

export default ApplePayButton
