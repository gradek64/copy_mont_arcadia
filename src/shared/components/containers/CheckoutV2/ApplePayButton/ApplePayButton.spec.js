import testComponentHelper from 'test/unit/helpers/test-component'
import ApplePayButton from './ApplePayButton'
import Button from '../../../common/Button/Button'

const renderComponent = testComponentHelper(ApplePayButton)

describe('<ApplePayButton/>', () => {
  describe('@renders', () => {
    it('the ApplePay button', () => {
      const { getTree } = renderComponent({})

      expect(getTree()).toMatchSnapshot()
    })

    it('passess props to the button component', () => {
      const props = {
        isActive: 'isActive',
        isDisabled: 'isDisabled',
        clickHandler: 'clickHandler',
      }
      const { wrapper } = renderComponent(props)

      expect(wrapper.find(Button).props()).toEqual(
        expect.objectContaining(props)
      )
    })
  })
})
