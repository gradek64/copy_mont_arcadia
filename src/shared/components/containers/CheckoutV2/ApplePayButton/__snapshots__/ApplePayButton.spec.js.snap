// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApplePayButton/> @renders the ApplePay button 1`] = `
<div
  className="ApplePayButton"
>
  <Button
    className="ApplePayButton--button"
  >
    <span
      className="text"
    >
      Buy with
    </span>
    <span
      className="logo"
    />
  </Button>
</div>
`;
